### PR TITLE
Feature/cloud 1161 add node group

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

TF module for creating node groups

Fixes [#2](https://usxtech.atlassian.net/browse/CLOUD-1161)

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create terraform.tfvars.json (provided below)
  - Create provider.tf (provided below)
  - Create version.tf (provided below)
  - Create main.tf (provided below)

Test Case 1 [ To create node group]:
- Steps
  -  I have used devops-playground cluster to test creation of node group.
  - Terraform validate
  - Terraform init
  - Terraform plan/apply
- Result
 - Node group will be created in dpl

 ```

## Examples

To test as a module added it here also in examples section of the root directory.

Set AWS profile as below before running the module

```bash
export AWS_PROFILE=dpl
```

versions.tf

```bash
terraform {
  required_version = ">=1.0.0"
  required_providers {
    eksctl = {
      source  = "mumoshu/eksctl"
      version = "0.16.2"
    }
  }
}
```

main.tf

```bash

module "node_group" {
  #source = "git::https://github.com/variant-inc/terraform-eksctl-nodegroup.git?ref=v1"
  #For branch
  source           = "git::https://github.com/variant-inc/terraform-eksctl-nodegroup.git?ref=feature/CLOUD-1161-add-node-group"
  name             = var.name
  region           = var.region
  cluster_name     = var.cluster_name
  managed          = var.managed
  enable_ssm       = var.enable_ssm
  nodes_min        = var.nodes_min
  nodes_max        = var.nodes_max
  nodes            = var.nodes
  instance_types   = var.instance_types
  node_volume_size = var.node_volume_size
  user_tags        = var.user_tags
  octopus_tags     = var.octopus_tags
}


```

.tfvars.json

```json
{
  "name": "ng-1",
  "cluster_region": "us-east-1",
  "cluster_name": "devops-playground",
  "managed": false,
  "nodes_min": 1,
  "nodes_max": 1,
  "nodes": 1,
  "instance_types": [
    "t3.micro"
  ],
  "node_volume_size": 10,
  "node_labels": {
    "autoscaling": "enabled",
    "iaac": true

  },
  "user_tags": {
    "team": "platform-engineering",
    "purpose": "node group test",
    "owner": "platform-engineering"
  },
  "octopus_tags": {
    "project": "test",
    "space": "Default",
    "environment": "development",
    "project_group": "Default Project Group",
    "release_channel": "release"
  },
  "node_taints":[{
    "key": "iaac11",
    "value": "true",
    "effect": "NO_SCHEDULE"
  }]
}
```

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

**Comment by vibindaniel** [comment]
Naveen please add pre-commit files as done in other terraform repositories. If you don't know them, we can have a call.

**Comment by naveen-variant** [comment]
> Naveen please add pre-commit files as done in other terraform repositories. If you don't know them, we can have a call.

updated as mentioned.

**Comment by salyateem13** [comment]
Please add an example in `/examples`

**Comment by naveen-variant** [comment]
> Please add an example in `/examples`

Added examples to the root directory.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,70 @@
...
+  - [Resources](#resources)
+  - [Inputs](#inputs)
+  - [Outputs](#outputs)
+
+## Usage
+
+```terraform
+  source = "github.com/variant-inc/terraform-eksctl-nodegroup?ref=v1"
+```
+
```
Comment (salyateem13) [inline README.md:22]: Add requirement to export AWS_PROFILE env variable for eksctl provider

File: outputs.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+output "kubeconfig_path" {
```
Comment (salyateem13) [inline outputs.tf:1]: Why is this output called kubeconfig_path

File: README.md (hunk 1 above)
Comment (naveen-variant) [reply README.md:23]: Added.

File: outputs.tf (hunk 2 above)
Comment (naveen-variant) [reply outputs.tf]: Updated the name

**Review by salyateem13 — APPROVED** [review]

File: versions.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">=1.0.0"
+  required_providers {
+    eksctl = {
+      source  = "mumoshu/eksctl"
+      version = "0.16.2"
```
Comment (vibindaniel) [inline versions.tf:6]: >=0.16.2

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,25 @@
+module "tags" {
+  source       = "github.com/variant-inc/lazy-terraform//submodules/tags?ref=v1"
+  name         = var.name
+  user_tags    = var.user_tags
+  octopus_tags = var.octopus_tags
+}
+
+resource "eksctl_nodegroup" "ng" {
```
Comment (vibindaniel) [inline main.tf:8]: Where are the labels?

Comment (vibindaniel) [reply main.tf:8]: Without labels autoscaling won't work

**Review by vibindaniel — CHANGES_REQUESTED** [review]
@naveen-variant Please go through the eksctl template used in iaac-eks and add all the fields mentioned there. I see 

disableIMDSv1
labels
taints

not added


disableIMDSv1 - Default is false, updating it to true will not allow us to set other iam parameters.
labels - Added
taints- There seems to be no support to add taints from eksctl nodegroup command. Tried checking in  eksctl create nodegroup -h command.

File: versions.tf (hunk 3 above)
Comment (naveen-variant) [reply versions.tf:6]: version parameter is not supporting decimal values. Have to be a string literal.

File: main.tf (hunk 4 above)
Comment (naveen-variant) [reply main.tf:8]: Added labels parameter

File: versions.tf (hunk 3 above)
Comment (vibindaniel) [reply versions.tf:6]: `>=0.16.2

**Review by salyateem13 — APPROVED** [review]

**Pull request merged**